dist/flatpak/com.github.input-leap.input-leaps.yml

```diff
@@ -10,34 +10,57 @@
 # https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1013
 
 app-id: com.github.inputleap.InputLeap
-runtime: org.gnome.Platform
-runtime-version: '42'
-sdk: org.gnome.Sdk
-command: input-leap-flatpak
+runtime: org.kde.Platform
+runtime-version: '6.7'
+sdk: org.kde.Sdk
+command: input-leap
 
 finish-args:
     # We want to talk to the network
     - --share=network
     # Need to read our config from somewhere
     - --filesystem=xdg-config/input-leap
-    # HACK: so we can put our EIS socket somewhere accessible
-    # This won't be needed once the portal actually supports our interfaces
-    - --filesystem=xdg-run/eis-0
-    # Let's show off :)
-    - --nosocket=x11
-    - --nosocket=wayland
+    - --socket=x11
+    - --socket=wayland
+    - --device=dri           # For hardware acceleration if needed
 
 modules:
+    - name: python3-attrs
+      buildsystem: simple
+      build-commands:
+        - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+          --prefix=${FLATPAK_DEST} --no-build-isolation attrs
+      sources:
+        - type: file
+          url: https://files.pythonhosted.org/packages/e0/44/827b2a91a5816512fcaf3cc4ebc465ccd5d598c45cefa6703fcf4a79018f/attrs-23.2.0-py3-none-any.whl
+          sha256: 99b87a485a5820b23b879f04c2305b44b951b502fd64be915879d77a7e8fc6f1
+      cleanup:
+        - /lib/python*
+    - name: python3-Jinja2
+      buildsystem: simple
+      build-commands:
+        - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+          --prefix=${FLATPAK_DEST} --no-build-isolation Jinja2
+      sources:
+        - type: file
+          url: https://files.pythonhosted.org/packages/30/6d/6de6be2d02603ab56e72997708809e8a5b0fbfee080735109b40a3564843/Jinja2-3.1.3-py3-none-any.whl
+          sha256: 7d6d50dd97d52cbc355597bd845fabfbac3f551e1f99619e39a35ce8c370b5fa
+        - type: file
+          url: https://files.pythonhosted.org/packages/87/5b/aae44c6655f3801e81aa3eef09dbbf012431987ba564d7231722f68df02d/MarkupSafe-2.1.5.tar.gz
+          sha256: d283d37a890ba4c1ae73ffadf8046435c76e7bc2247bbb63c00bd1a709c6544b
+      cleanup:
+        - /lib/python*
+  
     - name: libei
```

Comment:
Do libei, libportal etc have to be build from git? Can't we use the preexisting libraries?

Reply:
My not very deep understanding is that you select a base SDK, in this case I chose org.kde.Sdk/6.7. Then you have to build everything else. It seems some people want to have multiple base SDKs as discussed [here](https://github.com/flatpak/flatpak/issues/4797). I could be wrong however, and would love for someone more knowledgeable to comment.

Reply:
That's a good question though.

Looking at the [freedesktop SDK](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/) on which the KDE (and GNOME) SDK is based, it contains libei:

https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/elements/components/libei.bst?ref_type=heads

but not libportal though.

Reply:
That is fantastic that we will don't have to worry about libei. 
I am not sure but I think the portal stuff is handed by the flatpak "system".

Reply:
I'm thinking we leave it as is for now. Can I resolve this thread?

Reply:
>  I am not sure but I think the portal stuff is handed by the flatpak "system".

the "portal" (`xdg-desktop-portal`) is a process running outside the sandbox that provides a bunch of DBus apis that you can talk to. *lib*portal is a library that provides a GLib-like API to talk to those dbus interfaces - that one you need in your process and it is thus inside the sandbox.

I've mentioned this before elsewhere though that the long-term plan should be to replace the use of libportal in input-leap with QtDBus - the only reason we used libportal was because at the time @ofourdan and I were working on 6 repos at the same time to get this all in place and learning QtDBus on top was a bit... urgh.